pages/signup/register.tsx

```diff
@@ -1,60 +1,139 @@
+      ),
+    firstName: Yup.string().required(t("input-validation.required", { ns: "common" })),
+    lastName: Yup.string().required(t("input-validation.required", { ns: "common" })),
+    password: Yup.string()
+      .required(t("input-validation.required", { ns: "common" }))
+      .min(8, t("signUpRegistration.fields.passwordConfirmation.error.minLength"))
+      .max(50, t("signUpRegistration.fields.passwordConfirmation.error.maxLength"))
+      .test(
+        "password-valid-lowerCase",
+        t("signUpRegistration.fields.passwordConfirmation.error.oneLowerCase"),
+        (password = "") => isLowerCase(password)
+      )
+      .test(
+        "password-valid-upperCase",
+        t("signUpRegistration.fields.passwordConfirmation.error.oneUpperCase"),
+        (password = "") => isUpperCase(password)
+      )
+      .test(
+        "password-valid-number",
+        t("signUpRegistration.fields.passwordConfirmation.error.oneNumber"),
+        (password = "") => isNumber(password)
+      )
+      .test(
+        "password-valid-symbol",
+        t("signUpRegistration.fields.passwordConfirmation.error.oneSymbol"),
+        (password = "") => isSymbol(password)
+      ),
+    passwordConfirmation: Yup.string()
+      .required(t("input-validation.required", { ns: "common" }))
+      .oneOf(
+        [Yup.ref("password"), null],
+        t("signUpRegistration.fields.passwordConfirmation.error.mustMatch")
+      ),
+  });
+
   if (username) {
     return <Confirmation username={username} />;
   }
   return (
     <Formik
-      initialValues={{ username: "", password: "" }}
+      initialValues={{ username: "", password: "", firstName: "", lastName: "" }}
       onSubmit={async (values, formikHelpers) => {
         await register(values, formikHelpers, setUsername);
       }}
+      validateOnChange={false}
+      validateOnBlur={false}
+      validationSchema={validationSchema}
     >
-      {({ handleSubmit }) => (
+      {({ handleSubmit, errors, touched }) => (
         <>
           <h1>{t("signUpRegistration.title")}</h1>
-          <form id="registration" method="POST" onSubmit={handleSubmit}>
+          <form id="registration" method="POST" onSubmit={handleSubmit} noValidate>
+            {Object.keys(errors).length > 0 && Object.keys(touched).length > 0 ? (
+              <Alert
+                type="error"
+                heading={t("input-validation.heading", { ns: "common" })}
+                validation={true}
+                id={"validationAlert"}
+                tabIndex={0}
+              >
+                <ol className="gc-ordered-list">
+                  {errors &&
+                    Object.entries(errors).map(([elementId /*, errorMessage*/]) => {
+                      return (
+                        <ErrorListItem
+                          errorKey={elementId}
+                          value={t("input-validation.required", { ns: "common" })}
```

Comment:
Is there any reason we're not passing in the errorMessage here ?

Reply:
Completely missed that. Good point, will update